diff --git a/flow/scripts/synth_preamble.tcl b/flow/scripts/synth_preamble.tcl
index ae082367..13314dbb 100644
--- a/flow/scripts/synth_preamble.tcl
+++ b/flow/scripts/synth_preamble.tcl
@@ -1,4 +1,6 @@
 yosys -import
+plugin -i systemverilog
+yosys -import
 
 if {[info exist ::env(CACHED_NETLIST)]} {
   exec cp $::env(CACHED_NETLIST) $::env(RESULTS_DIR)/1_1_yosys.v
@@ -18,10 +20,8 @@ if {[info exist ::env(VERILOG_INCLUDE_DIRS)]} {
 }
 
 
-# Read verilog files
-foreach file $::env(VERILOG_FILES) {
-  read_verilog -defer -sv {*}$vIdirsArgs $file
-}
+# Read SystemVerilog files
+  read_systemverilog {*}$vIdirsArgs {*}$::env(VERILOG_FILES)
 
 
 
